src/box/memtx_tx.c

```diff
@@ -832,7 +832,7 @@ memtx_tx_story_delete(struct memtx_story *story)
 	assert(pos != mh_end(txm.history));
 	mh_history_del(txm.history, pos, 0);
 
-	story->tuple->is_dirty = false;
+	tuple_set_flags(story->tuple, story->tuple->flags & ~TUPLE_IS_DIRTY);
```

Comment:
It looks inconsistent. You use
- `tuple_is_dirty(tuple)` to check if `TUPLE_IS_DIRTY` is set;
- `tuple_set_flags(tuple, tuple->flags | TUPLE_IS_DIRTY);` to set the flag;
- `tuple_set_flags(story->tuple, story->tuple->flags & ~TUPLE_IS_DIRTY);` to clear the flag.

Also, it'd be nice to have a helper to clear a flag, rather than use bit-wise *not* and *and* operators directly.

Let's either
- add convenience helpers for testing/setting/clearing each flag:
  ```C
  tuple_is_dirty(tuple);
  tuple_set_dirty(tuple);
  tuple_clear_dirty(tuple);
  ```
- or add functions for testing/setting/clearing any flag:
  ```C
  tuple_has_flag(tuple, TUPLE_IS_DIRTY);
  tuple_set_flag(tuple, TUPLE_IS_DIRTY);
  tuple_clear_flag(tuple, TUPLE_IS_DIRTY);
  ```

Reply:
Refactored in the second way: added helpers `tuple_has_flag`, `tuple_set_flag`, `tuple_clear_flag`.